src/keystore/InMemoryKeystore.ts

```diff
@@ -254,6 +265,41 @@ export default class InMemoryKeystore implements Keystore {
     }
   }
 
+  async signDigest(
+    req: keystore.SignDigestRequest
+  ): Promise<signature.Signature> {
+    if (!validateObject(req, ['digest'], [])) {
+      throw new KeystoreError(
+        ErrorCode.ERROR_CODE_INVALID_INPUT,
+        'missing required field'
+      )
+    }
+
+    const { digest, identityKey, prekeyIndex } = req
+    let key: PrivateKey
+    if (identityKey) {
+      key = this.v1Keys.identityKey
+    } else if (
+      typeof prekeyIndex !== 'undefined' &&
+      Number.isInteger(prekeyIndex)
+    ) {
+      key = this.v1Keys.preKeys[prekeyIndex]
+      if (!key) {
+        throw new KeystoreError(
+          ErrorCode.ERROR_CODE_NO_MATCHING_PREKEY,
+          'no prekey found'
+        )
+      }
+    } else {
```

Comment:
My understanding from some of the previous discussion is that we really want to make this "sign any arbitrary digest" a legacy thing, and move towards signatures that are explicitly tied to a context. I would think that's going to be a new API, where it might make more sense to have explicit signing procedures (for example, if I'm signing a `preKey` the signer MUST be an `identityKey`)

Reply:
I do kinda like the idea of it being an enum with a fixed set of values. Definitely not too late to change the API for this one.